[16.0][FIX] l10n_es_aeat_mod303: comunicar IBAN en ingresos

yajo opened the pull request:
Cuando el 303 sale a ingresar, si el usuario indica una cuenta bancaria SEPA, se puede comunicar para domiciliarlo automáticamente.

La letra correspondiente al tipo de resultado sería la "U" (domiciliación del ingreso en CCC), pero resulta que esa letra nunca se usaba. Al parecer, al modelo le faltaba el concepto de [cuenta corriente tributaria][1], así que lo he añadido también. Esta cuenta corriente, a pesar del nombre, no es una cuenta bancaria; es un instrumento fiscal para gestionar la deuda de forma más ágil.

Por lo tanto, cuando el usuario escoge esa opción, la opción de cuenta bancaria desaparece y la marca SEPA se vuelve 0. También cambia el tipo a "V" cuando es a devolver.

Por ahora dejo el PR en borrador hasta que lo probemos mejor, para asegurarme de que funciona bien.

[1]: https://sede.agenciatributaria.gob.es/Sede/procedimientoini/GZ31.shtml?faqId=9767d0c33063c710VgnVCM100000dc381e0aRCRD

Fix https://github.com/OCA/l10n-spain/issues/3514
@moduon MT-5662

loida-vm commented:
Revisión funcional para el tipo de resultado "I" y "D". Ahora incluye en el archivo .txt la cuenta bancaria, para que se pueda realizar la domiciliación. 

(A falta de revisar el caso del check cuenta corriente tributaria)

loida-vm commented:
Revisión funcional para el tipo de resultado "I" y "D". Ahora incluye en el archivo .txt la cuenta bancaria, para que se pueda realizar la domiciliación.

(A falta de revisar el caso del check cuenta corriente tributaria)

yajo commented:
@pedrobaeza ya hemos usado este PR para presentar un 303 real y funciona como se espera, corrigiendo https://github.com/OCA/l10n-spain/issues/3514.

pedrobaeza commented:
¿Lo habéis hecho con la opción de cuenta tributaria? En esa opción diría que no hay que informar de ningún IBAN.

yajo commented:
No, lo hemos hecho con opción de cuenta bancaria SEPA (result_type=U).

pedrobaeza commented:
Ya, por eso. ¿No debería dejar fuera este PR entonces lo de la cuenta tributaria si no se ha probado? También creo que debería venir por separado cada cambio para mejor comprensión en caso de dejar ambos. Y eso, asegurar que el formato sería correcto para ese caso de la cuenta tributaria.

yajo commented:
No hay problema. Quito todo lo de la cuenta tributaria.

Lástima, fueron bastantes horas para añadirlo y "probablemente" esté bien... bueno, dejo aquí [el parche actual](https://github.com/OCA/l10n-spain/files/14957342/3515.patch.txt) por si hace falta recuperarlo en el futuro. 🤷🏼‍♂️

pedrobaeza commented:
OK, haz una cosa:

- Separa los dos cambios en 2 commits.
- Añade en el ROADMAP que no se ha probado lo de la cuenta tributaria.

yajo committed:
[FIX] l10n_es_aeat_mod303: comunicar IBAN en ingresos

Cuando el 303 sale a ingresar, si el usuario indica una cuenta bancaria SEPA, se puede comunicar para domiciliarlo automáticamente.

La letra correspondiente al tipo de resultado sería la "U" (domiciliación del ingreso en CCC), pero resulta que esa letra nunca se usaba.

Ahora el resultado tipo "I" se convierte en "U" si el usuario indica cuenta bancaria.

Fix https://github.com/OCA/l10n-spain/issues/3514
@moduon MT-5662

Co-authored-by: <104626797+loida-vm@users.noreply.github.com>

yajo committed:
[FIX] l10n_es_aeat_mod303: soporte para cuenta tributaria

Al modelo le faltaba el concepto de [cuenta corriente tributaria][1], así que lo he añadido. Esta cuenta corriente, a pesar del nombre, no es una cuenta bancaria; es un instrumento fiscal para gestionar la deuda de forma más ágil.

Por lo tanto, cuando el usuario escoge esa opción, la opción de cuenta bancaria desaparece y la marca SEPA se vuelve 0. También cambia el tipo a "V" cuando es a devolver.

[1]: https://sede.agenciatributaria.gob.es/Sede/procedimientoini/GZ31.shtml?faqId=9767d0c33063c710VgnVCM100000dc381e0aRCRD

@moduon MT-5662

yajo commented:
OK, buena idea. Ya lo tienes.

pedrobaeza reviewed (approved):
Gracias. ¿Podéis pasarlo a v17?

/ocabot merge minor
